chalice/awsclient.py

```diff
@@ -188,6 +236,120 @@ def create_function(self,
             kwargs['Layers'] = layers
         return self._create_lambda_function(kwargs)
 
+    def create_base_path_mapping(self,
+                                 domain_name,  # type: str
+                                 base_path,    # type: str
+                                 api_id,       # type: str
+                                 stage         # type: str
+                                 ):
+        # type: (...) -> Dict[str, str]
+        kwargs = {
+            'DomainName': domain_name,
+            'ApiMappingKey': base_path,
+            'ApiId': api_id,
+            'Stage': stage
+        }
+        return self._create_base_path_mapping(kwargs)
+
+    def create_domain_name(self,
+                           protocol,                       # type: str
+                           domain_name,                    # type: str
+                           endpoint_configuration,         # type: StrAnyMap
+                           security_policy,                # type: str
+                           hosted_zone_id=None,            # type: OptStr
+                           certificate_arn=None,           # type: str
+                           regional_certificate_arn=None,  # type: str
+                           tags=None,                      # type: StrMap
+                           ):
+        # type: (...) -> Dict[str, Any]
+        if protocol == 'HTTP':
+            kwargs = {
+                'domainName': domain_name,
+                'endpointConfiguration': endpoint_configuration,
+                'securityPolicy': security_policy,
+                'tags': tags
+            }
+            if certificate_arn:
+                kwargs['certificateArn'] = certificate_arn
+            if regional_certificate_arn:
+                kwargs['regionalCertificateArn'] = regional_certificate_arn
+            created_domain_name = self._create_domain_name(kwargs)
+        elif protocol == 'WEBSOCKET':
+            kwargs = self.get_custom_domain_params(
+                domain_name,
+                endpoint_configuration,
+                security_policy,
+                hosted_zone_id,
+                certificate_arn,
+                regional_certificate_arn,
+                tags
+            )
+            created_domain_name = self._create_domain_name_v2(kwargs)
+        else:
+            raise ValueError("Unsupported protocol value.")
+        return created_domain_name
+
+    def _create_base_path_mapping(self, api_args):
+        # type: (Dict[str, Any]) -> Dict[str, str]
+        result = self._client('apigatewayv2').create_api_mapping(**api_args)
```

Comment:
This doesn't seem to work when I try to use an EDGE endpoint type.  I get this error when I try to deploy my app:

```
 An error occurred (BadRequestException) when calling the CreateApiMapping
 operation: Only REGIONAL domain names can be managed through the API Gateway
 V2 API. For EDGE domain names, please use the API Gateway V1 API. Also note
 that only REST APIs can be attached to EDGE domain names.
```